exchange-scrapers/scrapers/HitBTCScraper.go

```diff
@@ -0,0 +1,221 @@
+package scrapers
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/diadata-org/api-golang/dia"
+	ws "github.com/gorilla/websocket"
+)
+
+var _socketurl string = "wss://api.hitbtc.com/api/2/ws"
+
+const WS_TIMEOUT = 10 * time.Second
+
+type Event struct {
+	Method string      `json:"method"`
+	Params interface{} `json:"params"`
+	Id     int         `json:"id"`
+}
+
+type HitBTCScraper struct {
+	wsClient *ws.Conn
+	// signaling channels for session initialization and finishing
+	//initDone     chan nothing
+	shutdown     chan nothing
+	shutdownDone chan nothing
+	// error handling; to read error or closed, first acquire read lock
+	// only cleanup method should hold write lock
+	errorLock sync.RWMutex
+	error     error
+	closed    bool
+	// used to keep track of trading pairs that we subscribed to
+	pairScrapers map[string]*HitBTCPairScraper
+}
+
+// NewHitBTCScraper returns a new HitBTCScraper for the given pair
+func NewHitBTCScraper(apiKey string, secretKey string) *HitBTCScraper {
+
+	s := &HitBTCScraper{
+		shutdown:     make(chan nothing),
+		shutdownDone: make(chan nothing),
+		pairScrapers: make(map[string]*HitBTCPairScraper),
+		error:        nil,
+	}
+
+	var wsDialer ws.Dialer
+	SwConn, _, err := wsDialer.Dial(_socketurl, nil)
+
+	if err != nil {
+		println(err.Error())
+	}
+
+	s.wsClient = SwConn
+	go s.mainLoop()
+	return s
+}
+
+// runs in a goroutine until s is closed
+func (s *HitBTCScraper) mainLoop() {
+
+	for true {
+
+		message := &Event{}
+		if err := s.wsClient.ReadJSON(&message); err != nil {
+			println(err.Error())
+			break
+		}
+
+		if message.Method == "updateTrades" {
+
+			md := message.Params.(map[string]interface{})
+			ps, ok := s.pairScrapers[md["symbol"].(string)]
+
+			if ok {
+
+				md_data := md["data"].([]interface{})
+
+				for _, v := range md_data {
+
+					md_element := v.(map[string]interface{})
```

Comment:
Why not using a library like https://github.com/bitbandi/go-hitbtc ?

Reply:
I have never seen it before, I have just looked in the repo and it seems to support only REST API, not WS.

Reply:
OK then let's keep your solution.